handlers/logout.go

```diff
@@ -27,17 +38,48 @@ func LogoutHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	sessstore.MaxAge(300)
 
-	var requestedURL = r.URL.Query().Get("url")
-	if requestedURL != "" {
+	providerLogoutURL := cfg.GenOAuth.LogoutURL
+	redirectURL := r.URL.Query().Get("url")
+
+	// Make sure that redirectURL, if given, is allowed by config
+	if redirectURL != "" {
+		redirectValid := false
 		for _, allowed := range cfg.Cfg.LogoutRedirectURLs {
-			if allowed == requestedURL {
+			if allowed == redirectURL {
 				log.Debugf("/logout found ")
-				redirect302(w, r, allowed)
-				return
+				redirectValid = true
+				break
 			}
 		}
-		error400(w, r, fmt.Errorf("%w: %s", errUnauthRedirURL, requestedURL))
-		return
+		if (!redirectValid) {
+			error400(w, r, fmt.Errorf("%w: %s", errUnauthRedirURL, redirectURL))
+			return
+		}
+	}
+
+	// If provider logout URL is configured, redirect to it (and pass redirectURL along)
+	// If provider logout URL is not configured, redirect directly to redirectURL
+	if providerLogoutURL != "" {
+		req, err := http.NewRequest("GET", providerLogoutURL, nil)
```

Comment:
Instead of using `http.NewRequest` please use `url.Parse` similar to this implementation...
https://github.com/coreos/go-oidc/pull/190/files

Reply:
Done